stratosphere/ams_mitm/source/fs_mitm/fsmitm_service.cpp

```diff
@@ -264,9 +265,15 @@ Result FsMitmService::OpenBisStorage(Out<std::shared_ptr<IStorageInterface>> out
             if (bis_partition_id == BisStorageId_Boot0) {
                 storage = std::make_shared<IStorageInterface>(new Boot0Storage(bis_storage, this->title_id));
             } else if (bis_partition_id == BisStorageId_Prodinfo) {
+                bool has_blank_cal0_flag = true; // See PR details
+
                 /* PRODINFO should *never* be writable. */
                 if (is_sysmodule || has_cal0_read_flag) {
                     storage = std::make_shared<IStorageInterface>(new ROProxyStorage(bis_storage));
+                } else if (has_blank_cal0_flag) {
```

Comment:
Wrong if ordering -- this will just result in this never doing the thing you want.

Reply:
But storage could just be reassinged twice, that was my reasoning..?

Reply:
this is in an else if, this ordering is wrong.

Reply:
oh, was thinking it was just another if block. oops